.github/workflows/docker-image.yml

```diff
@@ -20,26 +22,43 @@ jobs:
 
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v3
+      uses: actions/checkout@v3.0.2
+      with:
+        fetch-depth: 0
+
+    - name: Lower-case organization name
+      run: |
+        echo "ORGANIZATION=${ORGANIZATION,,}" >>${GITHUB_ENV}
 
     - name: Log in to the Container registry
-      uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+      uses: docker/login-action@v2.0.0
       with:
         registry: ${{ env.REGISTRY }}
-        username: ${{ github.actor }}
+        username: ${{ env.ORGANIZATION }}
         password: ${{ secrets.GITHUB_TOKEN }}
 
-    - name: Extract metadata (tags, labels) for Docker
-      id: meta
-      uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+    - name: Generate version number
+      id: semver
+      uses: paulhatch/semantic-version@v4.0.2
+      with:
+        major_pattern: "/^MAJOR:.*$/"
+        minor_pattern: "/^MINOR:.*$/"
+        tag_prefix: "v"
+        format: "${major}.${minor}.${patch}"
+        namespace: ${{ env.GITHUB_SHA }}
+
+    - name: Create GitHub release
```

Comment:
Do we really want a new GitHub release for every push on main? :D I can already feel the email spam.